From d17443e050a1ece6c3c5457e17b891bf0b772422 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 20 Jun 2021 23:33:07 +0300
Subject: [PATCH] [SW-46313] gaudi: security: prepare for sf cfg dyn

Update the HBW static RRs configuration to the new requirement - to have
the ability to open the SPI flash for reading through the SPI flash
memory mapped region.
This is a preparation for adding the dynamic configuration.

Change-Id: I1c3cb4f7091b0b631da6f7fe2f1d3ab698c07331
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/common.h b/arch/gaudi/common.h
index e932214..6cba60b 100644
--- a/arch/gaudi/common.h
+++ b/arch/gaudi/common.h
@@ -24,6 +24,7 @@
 
 #define DRAM_BASE			0x0
 #define HL_SPI_BASE			0x7FF8000000ull
+#define HL_SPI_SIZE			0x2000000
 #define HL_CFG_BASE			0x7FFC000000ull
 #define SDRAM_BASE			0x7FF0000000ull
 
diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index c9cdc93..eb1929e 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -1521,6 +1521,8 @@
 #define HBM_RR_BASE	0
 #define HBM_RR_MASK	0x3FFF800000000
 
+#define HL_SPI_END	(HL_SPI_BASE + HL_SPI_SIZE)
+
 static void preboot_secure_hbw(void)
 {
 	/* SRAM RR 0x7F_F000_0000 - 0x7F_F140_0000 (20MB) */
@@ -1555,12 +1557,13 @@
 	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
 	 * STM, SPI Flash, Scratchpad SRAM, PCIe FW SRAM.
 	 * SPSRAM space for fit is used in dynamic RR.
-	 * Assign the PCIe IF RR_16 , RR_23 (0-15 and 28-31 are allocated for
-	 * LBW) for this job.
+	 * Assign the PCIe IF RR_16, RR_22, RR_24 (0-15 and 28-31 are allocated
+	 * for LBW) for this job.
 	 * NOTE: The PCIe RRs use min/max concept and HIT is: min < hit < max
 	 */
-	preboot_secure_pcie_rr(16, mmSTM_BASE - 1, FIT_IMG_SPSRAM_BASE);
-	preboot_secure_pcie_rr(23, FIT_IMG_SPSRAM_END - 1, mmRAZWI_BASE);
+	preboot_secure_pcie_rr(PCIE_DYN_RR_STM_SF, mmSTM_BASE - 1, HL_SPI_END);
+	preboot_secure_pcie_rr(22, HL_SPI_END - 1, FIT_IMG_SPSRAM_BASE);
+	preboot_secure_pcie_rr(24, FIT_IMG_SPSRAM_END - 1, mmRAZWI_BASE);
 }
 
 static void preboot_security_pci_rtr_disable(int rr)
